enterprise/ha/src/main/java/org/neo4j/kernel/ha/com/master/ConversationManager.java

```diff
@@ -110,12 +127,22 @@ public void end( RequestContext context )
 
     public Set<RequestContext> getActiveContexts()
     {
-        return conversations.keys();
+        return conversations != null ? conversations.keys() : Collections.<RequestContext>emptySet() ;
     }
 
-    public void remove( RequestContext context )
+    /**
+     * Stop conversation for specified context.
+     * Conversation will still hold all already acquired locks, but will release all waiters and it will be
+     * impossible to get new locks out of it.
+     * @param context - context for which conversation should be stopped
+     */
+    public void stop( RequestContext context )
     {
-        conversations.remove( context );
+        Conversation conversation = conversations.end( context );
+        if ( conversation != null && conversation.isActive() )
+        {
+            conversation.stop();
+        }
```

Comment:
@davidegrohmann lock in [`Conversation`](https://github.com/neo4j/neo4j/pull/5163/files#diff-f7503123f85ed52ab7156227c304016cR51) was introduced to prevent that case to happen